webhook_router/app.py

```diff
@@ -2,17 +2,94 @@
 #  See LICENSE file for licensing details.
 
 """Flask application which receives GitHub webhooks and logs those."""
-import json
 import logging
 
+import yaml
+from flask import Config as FlaskConfig
 from flask import Flask, request
 
+from webhook_router.mq import add_job_to_queue
 from webhook_router.validation import verify_signature
+from webhook_router.webhook import Job, JobStatus
+from webhook_router.webhook.label_translation import (
+    FlavorLabelsMapping,
+    InvalidLabelCombinationError,
+    labels_to_flavor,
+    to_labels_flavor_mapping,
+)
+from webhook_router.webhook.parse import ParseError, webhook_to_job
 
+SUPPORTED_GITHUB_EVENT = "workflow_job"
+GITHUB_EVENT_HEADER = "X-Github-Event"
 WEBHOOK_SIGNATURE_HEADER = "X-Hub-Signature-256"
 
-app = Flask(__name__)
-app.config.from_prefixed_env()
+app = Flask(__name__.split(".", maxsplit=1)[0])
+
+
+class ConfigError(Exception):
+    """Raised when a configuration error occurs."""
+
+
+def config_app(flask_app: Flask) -> None:
+    """Configure the application.
+
+    Args:
+        flask_app: The Flask application to configure.
+
+    Raises:
+        ConfigError: If there is a validation error
+    """
+    flask_app.config.from_prefixed_env()
+    is_valid, err_msg = _validate_configs_are_set(flask_app.config)
+    if not is_valid:
+        raise ConfigError(err_msg)
+    yaml_str = flask_app.config["FLAVOURS"]
+    try:
+        flavor_labels_mapping = yaml.safe_load(yaml_str)
+    except yaml.YAMLError as exc:
+        raise ConfigError("Invalid 'FLAVOURS' config. Invalid yaml.") from exc
+    if not isinstance(flavor_labels_mapping, list):
+        raise ConfigError(
+            "Invalid 'FLAVOURS' config. Expected a YAML file with a list at the top level."
+        )
+    flavor_labels_mapping = [tuple(item.items())[0] for item in flavor_labels_mapping]
+    try:
+        flavor_labels_mapping = FlavorLabelsMapping(mapping=flavor_labels_mapping)
+    except ValueError as exc:
+        raise ConfigError("Invalid 'FLAVOURS' config. Invalid format.") from exc
+    flask_app.config["LABEL_FLAVOR_MAPPING"] = to_labels_flavor_mapping(
+        flavor_labels_mapping,
+        ignore_labels=set(flask_app.config["GITHUB_DEFAULT_LABELS"].split(",")),
+    )
+
+
+def _validate_configs_are_set(config: FlaskConfig) -> tuple[bool, str]:
```

Comment:
Just out of curiosity, may I ask why we are not validating it in `flask_app.config.from_prefixed_env()`?
I am not against this pattern but I'm wondering if this might belong in `from_prefixed_env` config :)

Reply:
Can you please elaborate further, [`from_prefixed_env`](https://flask.palletsprojects.com/en/2.3.x/api/#flask.Config.from_prefixed_env) is a method owned by the Flask framework.

Reply:
Oh I see, thank you for the explanation!
Another question is, I'm wondering if there is a reason for doing validation for if configs are set and another to check the configs are correct?

Reply:
So you would prefer to remove the method and use `config.get` calls with a default value? I think one advantage is that we have a better error message, which makes life easier for the operator (it is easier to know that something is missing than to know that you have to fix something wrongly constructed).

Reply:
Oh sorry, I think I should've clarified:
I was wondering if other validations like YAML validations
```
try:
        flavor_labels_mapping = yaml.safe_load(yaml_str)
    except yaml.YAMLError as exc:
        raise ConfigError("Invalid 'FLAVOURS' config. Invalid yaml.") from exc
    if not isinstance(flavor_labels_mapping, list):
        raise ConfigError(
            "Invalid 'FLAVOURS' config. Expected a YAML file with a list at the top level."
        )
```
should also live in the `validate` function and have ` validation function rather than `validate_set` function.

Reply:
I have refactored the code to not include a validation function at all (but do the validation while parsing/constructing the object).